analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -345,6 +345,16 @@ def add_arguments_to_parser(parser):
                                     "'<OUTPUT_DIR>/ctu-dir'. (These files "
                                     "will not be cleaned up in this mode.)")
 
+        ctu_opts.add_argument('--ctu-on-demand',
```

Comment:
Does the user get any warnings if `--ctu --ctu-collect --ctu-on-demand` is used together?

Reply:
I have created a new option-group called ast-loading-mode, that way it will not conflict with the original ctu-options.